File: Project_2/springboot-backend/src/main/java/com/zombieapocalypse/springbootbackend/service/AccountService.java
package com.zombieapocalypse.springbootbackend.service;

import com.zombieapocalypse.springbootbackend.model.Account;
import java.util.List;

public interface AccountService {

    // Create CRUD function
    public Account createAccount(Account account);
    // Retrieve CRUD function
    public List<Account> getAccounts();
    // Retrieve CRUD function
    public Account getAccountById(int id);
    // Update: CRUD function
    public Account updateAccount(int id, Account account);
    // Delete: CRUD function
    public String deleteAccount(int id);
}

File: Project_2/Game/src/service/AccountService.js
import axios from 'axios';

const ACCOUNT_API_BASE_URL = 'http://localhost:8080/api/v1/accounts';

class AccountService {
  saveAccount(account) {
    return axios.post(ACCOUNT_API_BASE_URL, account);
  }

  getAccounts() {
    return axios.get(ACCOUNT_API_BASE_URL);
  }

  getAccountById(id) {
    return axios.get(ACCOUNT_API_BASE_URL + '/' + id);
  }

  updateAccount(account, id) {
    return axios.put(ACCOUNT_API_BASE_URL + '/' + id, account);
  }

  deleteAccount(id) {
    return axios.delete(ACCOUNT_API_BASE_URL + '/' + id);
  }
}

export default new AccountService();
